src/state/index.ts

```diff
@@ -21,27 +22,87 @@ import wallets from './wallets/reducer'
 
 const PERSISTED_KEYS: string[] = ['user', 'transactions', 'lists']
 
+const reducer = {
+  application,
+  user,
+  connection,
+  transactions,
+  wallets,
+  swap,
+  mint,
+  mintV3,
+  burn,
+  burnV3,
+  multicall: multicall.reducer,
+  lists,
+  logs,
+  [routingApi.reducerPath]: routingApi.reducer,
+}
+
+/* Utility type to extract state type out of a @reduxjs/toolkit Reducer type */
+type GetState<T> = T extends Reducer<infer State> ? State : never
+
+/* Utility type to mark all properties of a type as optional */
+type DeepPartial<T> = T extends object
+  ? {
+      [P in keyof T]?: DeepPartial<T[P]>
+    }
+  : T
+
+/* eslint-disable-next-line import/no-unused-modules -- reports false positive error, we use that type in other modules */
+export type AppState = {
+  [K in keyof typeof reducer]: GetState<typeof reducer[K]>
+}
+
+/* eslint-disable-next-line import/no-unused-modules -- reports false positive error, we use that type in other modules */
+export type AppDispatch = typeof store.dispatch
+
 const store = configureStore({
-  reducer: {
-    application,
-    user,
-    connection,
-    transactions,
-    wallets,
-    swap,
-    mint,
-    mintV3,
-    burn,
-    burnV3,
-    multicall: multicall.reducer,
-    lists,
-    logs,
-    [routingApi.reducerPath]: routingApi.reducer,
-  },
+  reducer,
   middleware: (getDefaultMiddleware) =>
     getDefaultMiddleware({ thunk: true })
       .concat(routingApi.middleware)
-      .concat(save({ states: PERSISTED_KEYS, debounce: 1000 })),
+      .concat(save({ states: PERSISTED_KEYS, debounce: 1000 }))
+      .concat(
+        Sentry.createReduxEnhancer({
+          actionTransformer: () => null,
+          /**
+           * This function runs on every state update, so keeping it as fast as possible by avoiding any function
```

Comment:
This comment should also explain _why_ this is being done.

Reply:
Added some more comments